kimchi/src/circuits/expr.rs

```diff
@@ -987,10 +984,10 @@ impl<'a, F: FftField> EvalResult<'a, F> {
                     evals: es_sub,
                 },
             ) => {
-                let scale = (d_sub as usize) / (d as usize);
-                assert!(scale != 0);
+                let es_sub_evals =
+                    to_domain(es_sub, d_sub as usize, d as usize, res_domain.1, s, None);
                 evals.evals.par_iter_mut().enumerate().for_each(|(i, e)| {
-                    *e += es_sub.evals[(scale * i + (d_sub as usize) * s) % es_sub.evals.len()];
+                    *e += es_sub_evals[i];
                 });
```

Comment:
I think we should think of a way to have that factored out as well, and as you said in https://github.com/o1-labs/proof-systems/issues/886#issuecomment-1337447961 not create this temporary unecessary vec

maybe a `to_domain_in_place` function?